components/fieldlabel/stories/fieldlabel.stories.js

```diff
@@ -39,15 +41,91 @@ export default {
 		alignment: "left",
 		isDisabled: false,
 		isRequired: false,
+		label: "Label",
 	},
 	parameters: {
 		packageJson: pkgJson,
 	},
 };
 
+/**
+ * By default, a field label has left-aligned text, and is a medium size. For field label text, use a short, catch-all description (1-3 words) of the information that a user needs to provide.
+ * 
+ * The default position of a field label is above an input, but it can alternatively be positioned to the side. Visit the [form component](/docs/components-form--docs) to see examples of the field label with an input.
+ 
+ */
 export const Default = FieldLabelGroup.bind({});
-Default.args = {
+Default.args = {};
+
+// ********* DOCS ONLY ********* //
+/**
+ * Field labels come in four different sizes: small, medium, large, and extra-large. The medium size is the default and most frequently used option with medium-sized inputs. Use the other sizes sparingly; they should be used to create a hierarchy of importance within the page. 
+ * 
+ * Both small and medium field labels have the same font size, but different paddings when used as side labels.
+ */
+export const Sizing = (args, context) => Sizes({
+	Template,
+	withBorder: false,
+	withHeading: false,
+	...args,
+}, context);
+Sizing.tags = ["!dev"];
+Sizing.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+
+/**
+ * To render right-aligned field label text, apply the `.spectrum-FieldLabel--right` class to the field label.
+ * 
+ */
+export const RightAligned = Template.bind({});
+RightAligned.args = {
 	label: "Label",
+	alignment: "right",
+	customStyles: {
+		width: "72px",
+	},
+};
+RightAligned.tags = ["!dev"];
+RightAligned.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+RightAligned.storyName = "Right-aligned";
+
+/**
+ * Field labels for required inputs can be marked with an asterisk at the end of the label. Optional inputs would then be understood to not have the asterisk. If using the asterisk icon, do not leave any space between the label text and the start of the `<svg>` element in the markup. Extra space should not be added in addition to the inline margin.
```

Comment:
There's a bit about the "necessity indicator" in the [Spectrum guidelines that says you can use text](https://spectrum.adobe.com/page/field-label/#Necessity-indicator-(required-or-optional)) like "(required)" or "(optional)". I didn't see that in the S2 field label Figma file, so I left it off for now.

If we want to add it, I certainly can. I believe I'd need to add an extra control when required is true, to swap between a text and icon option.